lib/web_ui/skwasm/text/paragraph.cpp

```diff
@@ -58,6 +61,46 @@ SKWASM_EXPORT int32_t paragraph_getPositionForOffset(Paragraph* paragraph,
   return position.position;
 }
 
+SKWASM_EXPORT bool paragraph_getClosestGlyphInfoAtCoordinate(
+    Paragraph* paragraph,
+    SkScalar offsetX,
+    SkScalar offsetY,
+    SkScalar* graphemeLayoutBounds,
+    size_t* garphemeCodeUnitRange,
+    bool* booleanFlags) {
```

Comment:
Instead of returning a boolean here, you can directly return a `skia::textlayout::TextDirection`, and we use that to index into the enum on the dart side. Check out `textBoxList_getBoxAtIndex` as an example of this.

Reply:
The return value currently indicates whether the out parameters are valid. To add `TextDirection` information to it I think we'd need to return tristate value (LTR, RTL, invalid)?

Reply:
Sorry, to clarify, my suggestion is to return it by reference as you're doing, but instead of using a `bool` just return a `TextDirection` by reference. You can leave the actual return value as a normal boolean, indicating success/failure of the overall API call.

Reply:
Ah I just realized the parameters are called "booleanFlags" because there are a few more booleans I'd like to expose after this (e.g., `isPlaceholder`, `isEllipsis`) that I'd like to pack together. Would it be ok to keep this a `bool*` for now?

Reply:
I'm fine with leaving it as-is for now. I think longer term maybe what we should be doing is actually declaring the C structs themselves on the dart side using ffi constructs and then actually just passing one struct (right by value or by reference) across the boundary. I have been meaning to experiment with that with these bindings but haven't really had much of a chance.